app/jobs/set_participant_completion_date_job.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# This is a one off job to update all of the induction completion dates from DQT for
+# all of the 2021/2022 ECTs that do not currently have them.
+# It runs in batches of 200 as we have a 300 lookup per minute limit on the API
+# This could get adapted later to be a regular running process with a bit more logic
+class SetParticipantCompletionDateJob < ApplicationJob
+  def perform
+    ParticipantProfile::ECT
+      .eligible_status
+      .joins(:teacher_profile)
+      .includes(:teacher_profile)
+      .where.not(teacher_profile: { trn: nil })
+      .where.not(induction_start_date: nil)
+      .where(created_at: ...Cohort.find_by(start_year: 2023).registration_start_date)
```

Comment:
why exclude ects registered this year? It will become invalid next year or the next one

Reply:
This is only for backfilling for now.